src/core/net/dnssd_server.cpp

```diff
@@ -867,6 +867,11 @@ void Server::OnUpstreamQueryDone(UpstreamQueryTransaction &aQueryTransaction, Me
         error = mSocket.SendTo(*aResponseMessage, aQueryTransaction.GetMessageInfo());
     }
 
+    if (error == kErrorNone)
+    {
+        mCounters.mHandledByUpstream.mForwardedResponses++;
+    }
```

Comment:
Update failed counter on `error` here? (we got a callback with response but could not send it out?).

Also if `aResponseMessage == nullptr` should we update failed counter? (can happen on a timeout).